### PR TITLE
Add `skuba start` fastify support

### PR DESCRIPTION
**Pull request by samchungy**
Tested on my upcoming template:

https://github.com/samchungy/test-fastify-template/pull/1

### CODE REVIEW

File: src/wrapper/requestListener.ts [hunk 1]
```diff
@@ -15,11 +15,16 @@ interface ObjectConfig {
...
+  server?: http.Server;
+
   default?: unknown;
   port?: unknown;
 }
 
-const isConfig = (data: unknown): data is FunctionConfig | ObjectConfig =>
+const isConfig = (
+  data: unknown,
+): data is Promise<FunctionConfig> | Promise<ObjectConfig> =>
```
Comment (72636c) [inline src/wrapper/requestListener.ts:27]: 🤔 Why are these promises?

**Review by 72636c — APPROVED** [review]

File: src/wrapper/requestListener.ts (hunk 1 above)
Comment (samchungy) [reply src/wrapper/requestListener.ts:27]: Fastify is a confusing hybrid of callbacks and promises. To support both styles we gotta make it a Promise.

https://www.fastify.io/docs/latest/Guides/Getting-Started/#your-first-server

Comment (samchungy) [reply src/wrapper/requestListener.ts:27]: image

Comment (72636c) [reply src/wrapper/requestListener.ts:27]: I don't think this guard is properly checking for a nested `Promise` type though. The input would have to be awaited before being passed in

Comment (samchungy) [reply src/wrapper/requestListener.ts:27]: https://github.com/samchungy/test-fastify-template/pull/1 if you wanna peek. Still very much a WIP. Putting down the tools for the day :)

Comment (samchungy) [reply src/wrapper/requestListener.ts:27]: The app.ts calls createApp() and exports it so it should be okay

Comment (samchungy) [reply src/wrapper/requestListener.ts:27]: Alright added some tests and better typings

**Review by 72636c — APPROVED** [review]
This looks great, thanks!

Up to you on whether you want to changeset this PR or leave it until you have more to share 🙂

**Pull request merged**